controllers/goharbor/trivy/deployments.go

```diff
@@ -0,0 +1,158 @@
+package trivy
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	goharborv1alpha2 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	"github.com/pkg/errors"
+)
+
+var (
+	varFalse = false
+)
+
+const (
+	ContainerName     = "trivy"
+	LivenessProbe     = "/probe/healthy"
+	ReadinessProbe    = "/probe/ready"
+	port              = 8080 // https://github.com/helm/chartmuseum/blob/969515a51413e1f1840fb99509401aa3c63deccd/pkg/config/vars.go#L135
+	CacheVolumeName   = "cache"
+	CacheVolumePath   = "/home/scanner/.cache/trivy"
+	ReportsVolumeName = "reports"
+	ReportsVolumePath = "/home/scanner/.cache/reports"
+)
+
+func (r *Reconciler) AddDeployment(ctx context.Context, trivy *goharborv1alpha2.Trivy) error {
+	// Forge the deploy resource
+	deploy, err := r.GetDeployment(ctx, trivy)
+	if err != nil {
+		return errors.Wrap(err, "cannot get deployment")
+	}
+
+	// Add deploy to reconciler controller
+	_, err = r.Controller.AddDeploymentToManage(ctx, deploy)
+	if err != nil {
+		return errors.Wrapf(err, "cannot manage deploy %s", deploy.GetName())
+	}
+
+	return nil
+}
+
+func (r *Reconciler) GetDeployment(ctx context.Context, trivy *goharborv1alpha2.Trivy) (*appsv1.Deployment, error) { // nolint:funlen
+	name := r.NormalizeName(ctx, trivy.GetName())
+	namespace := trivy.GetNamespace()
+
+	image, err := r.GetImage(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot get image for deploy: %s", name)
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: CacheVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: ReportsVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
```

Comment:
@danielpacak 

For trivy, temporary storage is totally fine, correct?

Reply:
Yes. That's fine. What's more it's required with K8S Deployment to avoid accessing shared `trivy.db` file by multiple `trivy` executable processes spawned by the adapter service.